### PR TITLE
feat: support decorated expressions

### PR DESCRIPTION
**Pull request by aspeddro**
See [reanalyze exampes](https://github.com/rescript-association/reanalyze/blob/388149046bf077d17639f50d4c7aea2bdd28b8b5/examples/deadcode/src/noalloc/Chess.res#L19-L20)

```res
@local
call()
```

```
(source_file [0, 0] - [2, 0]
  (ERROR [0, 0] - [1, 6]
    (decorator [0, 0] - [1, 6]
      (decorator_identifier [0, 1] - [0, 6])
      (ERROR [1, 0] - [1, 4]
        (ERROR [1, 0] - [1, 4]))
      (decorator_arguments [1, 4] - [1, 6]))))
```

### CODE REVIEW

File: grammar.js [hunk 1]
```diff
@@ -175,6 +176,7 @@ module.exports = grammar({
       $.module_declaration,
       $.external_declaration,
       $.exception_declaration,
+      $.call_expression,
```
Comment (nkrkv) [inline grammar.js:179]: Call _expression_ is definitely an _expression_, not a _declaration_.

I understand the intent, but it should be structured in another way I guess.

**Review by nkrkv — CHANGES_REQUESTED** [review]

File: grammar.js (hunk 1 above)
Comment (aspeddro) [reply grammar.js]: I moved to `call_expression` to `_decorated_statement`

File: grammar.js [hunk 2]
```diff
@@ -143,7 +143,10 @@ module.exports = grammar({
 
     _decorated_statement: $ => seq(
       repeat1($.decorator),
-      $.declaration,
+      choice(
+        $.declaration,
+        $.call_expression
```
Comment (aspeddro) [inline grammar.js:148]: Maybe we should use `expression_statement` here. Since decorators/attributes can appear anywhere.

**Review by nkrkv — APPROVED** [review]
It is fine now. Thank you!

Yes, decorators can appear anywhere basically. In that sense they might be implemented as comments, for example. But in this case they would appear as siblings to the parts of code they decorate and would not participate in, for example, broader/narrower selection. So I think it is better to eventually come to the grammar where all possible cases are covered semantically. Your addition improves the situation following this way :+1:

**Pull request merged**